api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricHandler.java

```diff
@@ -472,14 +352,54 @@ public void findNumericData(
         ListenableFuture<NumericMetric> dataFuture = metricsService.findNumericData(metric, start, end);
         ListenableFuture<? extends Object> outputFuture = null;
         if (numberOfBuckets == 0) {
-            outputFuture = Futures.transform(dataFuture, new MetricOutMapper());
+            outputFuture = Futures.transform(dataFuture, new Function<NumericMetric, List<NumericData>>() {
+
+                @Override
+                public List<NumericData> apply(NumericMetric metric) {
+                    if (metric == null) {
+                        throw new NoResultsException();
+                    }
+
+                    return metric.getData();
+                }
+            });
         } else {
             if (bucketWidthSeconds == 0) {
                 outputFuture = Futures.transform(dataFuture, new CreateSimpleBuckets(start, end, numberOfBuckets,
                     skipEmpty));
             } else {
                 ListenableFuture<List<? extends Object>> bucketsFuture = Futures.transform(dataFuture,
-                    new CreateFixedNumberOfBuckets(numberOfBuckets, bucketWidthSeconds));
+                        new Function<NumericMetric, List<? extends Object>>() {
```

Comment:
The code is complicated. I checked with Mike Thompson a while back, and most of it (at least at that time) was not actually used. We need to find out what is actually used, get rid of the rest, and then refactor. If it is not used, I don't want to keep it. It is just technical debt.